Update BSD68 experiment

diyasrivas opened the pull request:
fixes #9 

- [x] update python script
- [x] update readme
- [x] add slurm script
- [ ] test run

diyasrivas committed:
Update BSD68 experiment

jdeschamps commented on experiments/n2v/bsd68/careamics/README.md:
```diff
@@ -1,27 +1,33 @@
+)
+
+# Updated parameters from original script
+config["algorithm_config"]["model"]["num_channels_init"] = 32
```
The model parameters can be passed in `model_params` in `create_n2v_configuration` now.

diyasrivas replied:
Updated both README.md and careamics-bsd68.py to use model_params directly in create_n2v_configuration()

diyasrivas committed:
Remove Convallaria files - belongs in different branch

diyasrivas committed:
fixing bsd68 experiment files

diyasrivas commented on experiments/n2v/convallaria/careamics/careamics-convallaria.py:
```diff
@@ -0,0 +1,87 @@
+
+# Initialize and train model
+print("Initializing and training model...")
+working_dir = os.path.join(results_dir, "checkpoints")
```
to address in #3

diyasrivas commented on experiments/n2v/convallaria/careamics/careamics-convallaria.py:
```diff
@@ -0,0 +1,87 @@
+    num_epochs=100,
+    masked_pixel_percentage=0.2,
+    struct_n2v_axis="none",
+    model_checkpoint={},
```
to address in #3

diyasrivas commented on experiments/n2v/convallaria/careamics/careamics-convallaria.py:
```diff
@@ -0,0 +1,87 @@
+data_path.mkdir(parents=True, exist_ok=True)
+
+# Paths to data files
+conv_path = "/group/jug/datasets/Convallaria_diaphragm/20190520_tl_25um_50msec_05pc_488_130EM_Conv.tif"
```
to address in #3

jdeschamps commented on experiments/n2v/bsd68/careamics/README.md:
```diff
@@ -22,19 +21,21 @@ config = create_n2v_configuration(
+    model_params={
+        "num_channels_init": 32
+    },
+    optimizer_params={
```
The `optimizer_params` and `lr_scheduler_params` do not exist in the signature of `create_n2v_configuration`:

https://github.com/CAREamics/careamics/blob/0fa617d7e27c61005274e433a3422be97bcadd7d/src/careamics/config/configuration_factories.py#L748

While it would probably make sense to add them, this is currently not the case. You should test the code right away when making modifications!

jdeschamps replied:
Just to clarify further the `model_parameter` is a valid parameter, hence the original comment.

Opened an issue: https://github.com/CAREamics/careamics/issues/478

jdeschamps commented on experiments/n2v/bsd68/careamics/careamics-bsd68.py:
```diff
@@ -33,106 +29,70 @@
+    model_params={
+        "num_channels_init": 32
+    },
+    optimizer_params={
```
same here, `optimizer_params` and `lr_scheduler` are not valid parameters of that function! They probably should...

jdeschamps replied:
If you deal with the issue, then better not remove the call from here since you are about to fix it!

jdeschamps commented on experiments/n2v/bsd68/careamics/careamics-bsd68.py:
```diff
@@ -1,29 +1,25 @@
+root_path = Path("data")
 files = portfolio.denoising.N2V_BSD68.download(root_path)
 print(f"List of downloaded files: {files}")
 
```
Not this particular line, but I noticed that there is this line: 

```
pprint.PrettyPrinter(indent=2).pprint(config)
```

Which is probably useless and it was meant to make the configuration printing look nice in jupyter notebook.

diyasrivas replied:
must have missed it - removed

jdeschamps commented on experiments/n2v/bsd68/careamics/careamics-bsd68.py:
```diff
@@ -41,72 +39,60 @@
 
-print(f"PSNR total: {psnr_total / len(preds)}")
+print(f"Average PSNR: {psnr_total / len(predictions):.2f}")
+print(f"Average MicroSSIM: {microssim_total / len(predictions):.3f}")
```
let's leave it at two decimals!

diyasrivas committed:
fixing files

diyasrivas committed:
updated files after HPC testing

jdeschamps commented on experiments/n2v/bsd68/careamics/careamics-bsd68.py:
```diff
@@ -48,20 +47,10 @@
     num_epochs=50,
     masked_pixel_percentage=0.2,
     struct_n2v_axis="none",
-    model_params={
```
So `model_params` was correct, just the other two parameters were not existing (yet).

Now they are in CAREamics (since this morning): https://github.com/CAREamics/careamics/pull/483

Make sure that the optimizer and lr_scheduler are indeed the same as in the tensorflow implementation!

jdeschamps commented on experiments/n2v/bsd68/careamics/careamics-bsd68.py:
```diff
@@ -63,18 +62,27 @@
+
     pred = careamist.predict(
-        source=test_img,
+        source=test_file, 
```
Here you predict on `test_file` which is probably a `path`, but just before you load the image in `test_img`.

It probably makes sense to predict rather on the image you loaded, since you already have it in memory and need it for the metrics computation.

diyasrivas committed:
Fix scripts and optimize HPC runtime allocation

diyasrivas committed:
reading all images and change training to array

diyasrivas commented on experiments/n2v/bsd68/careamics/careamics-bsd68.py:
```diff
@@ -46,6 +46,17 @@
     num_epochs=50,
     masked_pixel_percentage=0.2,
     struct_n2v_axis="none",
+    logger="wandb", 
```
updated this, now logs show:
_Loaded 3168 training images
Loaded 4 validation images_

diyasrivas committed:
tidying up files

diyasrivas committed:
add stddev

diyasrivas committed:
fixed data paths issue

diyasrivas commented:
Cleaned up this branch, the problem was a previous commit 125f032 contained both BSD68 and Convallaria experiments (oops). ds/bsd68 kept BSD68 files but deleted Convallaria whilst the main branch retained Convallaria files (via PR #3) but doesn't have BSD68. I rebased and went through my previous commits to clean up ds/bsd68. There should be no conflicts with the base branch now ...

diyasrivas commented on experiments/n2v/bsd68/careamics/careamics-bsd68.py:
```diff
@@ -36,77 +32,93 @@
 test_path.mkdir(parents=True, exist_ok=True)
 gt_path.mkdir(parents=True, exist_ok=True)
 
-#### training and validation data
```
fixed
